packages/syntax-core/src/RichSelect/RichSelectBox.tsx

```diff
@@ -0,0 +1,222 @@
+  autoFocus?: boolean;
+};
+export const RichSelectBoxContext = createContext<RichSelectBoxContextType>({});
+
+const RichSelectBox = forwardRef<HTMLDivElement, RichSelectBoxProps>(
+  function RichSelectBox(props, ref): ReactElement {
+    const {
+      accessibilityLabel,
+      autosave,
+      children,
+      "data-testid": dataTestId,
+      multiple = false,
+      onChange,
+      primaryButtonText = "Save",
+      primaryButtonAccessibilityLabel = "Save",
+      secondaryButtonText = "Clear",
+      secondaryButtonAccessibilityLabel = "Clear",
+      selectedValues: selectedValuesProp,
+      defaultSelectedValues: defaultSelectedValuesProp,
+    } = props;
+
+    const selectedKeysProp = useMemo(
+      () => convertSelection(selectedValuesProp),
+      [selectedValuesProp],
+    );
+    const defaultSelectedKeys = useMemo(
+      () => convertSelection(defaultSelectedValuesProp, new Set()),
+      [defaultSelectedValuesProp],
+    );
+    const [selectedKeys, setSelectedKeys] = useControlledState(
+      selectedKeysProp!, // eslint-disable-line @typescript-eslint/no-non-null-assertion -- there is a bug in the typedef for useControlledState from react-stately.  Internally they rely on value (first arg) able to be undefined
+      defaultSelectedKeys,
+      (value) => {
+        if (isEqualSelection(value, selectedKeys)) return;
+        // Notify parent about the changes
+        if (value === "all") return onChange("all");
+        onChange([...value].map(String));
+      },
+    );
+    const [stagedKeys, setStagedKeys] = useState<Set<Key> | "all">(
+      selectedKeys,
+    );
+
+    const saveChanges = () => setSelectedKeys(stagedKeys);
+    const clearChanges = () => setStagedKeys(new Set());
+    const stageChanges = (selectedValues: Selection) => {
+      setStagedKeys(selectedValues);
+      if (autosave) setSelectedKeys(selectedValues);
+    };
+
+    // inject method into context so children can disable themselves
+    // by adding `disabled` attribute (through RichSelectItem)
+    const [disabledKeysComposed, setDisabledKeysComposed] = useState<Set<Key>>(
+      new Set(),
+    );
+    const disableKey = useCallback((key: Key, _disabled: boolean) => {
+      setDisabledKeysComposed((keys) => {
+        if (keys.has(key) === _disabled) return keys;
+        _disabled ? keys.add(key) : keys.delete(key);
+        return new Set(keys);
+      });
+    }, []);
+
+    // higher level context for autoFocus behavior (parent sets when rendering RichSelectBox in overlay)
+    const { autoFocus } = useContext(RichSelectBoxContext);
+
+    return (
+      <RichSelectItemContext.Provider value={{ disableKey }}>
+        <Box>
+          <ReactAriaListBox
+            ref={ref}
+            aria-label={accessibilityLabel}
+            autoFocus={autoFocus}
+            selectionMode={multiple ? "multiple" : "single"}
+            selectionBehavior={multiple ? "toggle" : "replace"}
```

Comment:
single mode autosave=true that bug we saw where it closed immediately on keyboard nav could be fixed by turning this to toggle

Reply:
nope, that wouldn't work.  instead, only autoclose the overlay when `autosave=false`: https://github.com/Cambly/syntax/pull/307/commits/cd27b6fb2fc7777787891888bec645b680ba3780